dev/src/codewind/project/Project.ts

```diff
@@ -464,6 +466,77 @@ export default class Project implements vscode.QuickPickItem {
         });
     }
 
+    public async onLoadRunnerUpdate(event: {projectID: string, status: string, timestamp: string}): Promise<void> {
+        Log.d(`${this.name} load runner status changed to ${event.status}`, event);
+        switch (event.status) {
+            case "hcdReady":
+            case "profilingReady":
+                Log.d("Profiling data is ready to be saved to workspace");
+                const loadTestPath = path.join(this.localPath.fsPath, "load-test");
+                const timestampPath = path.join(loadTestPath, event.timestamp);
+                let fileName = "";
+                if (this.language === "java") {
+                    fileName = "profiling.hcd";
+                } else if (this.language === "nodejs") {
+                    fileName = "profiling.json";
+                }
+                const profilingOutPath = path.join(timestampPath, fileName);
+                try {
+                    await fs.promises.mkdir(timestampPath, {recursive: true});
+                    const fd = await fs.promises.open(profilingOutPath, "w");
+                    await fd.close();
+                } catch (error) {
+                    Log.e(error);
+                    vscode.window.showErrorMessage(`Could not create file at ${profilingOutPath}`);
+                    return;
+                }
+                const endpoint = ProjectEndpoints.PROFILING.toString().concat(`/${event.timestamp}`);
+                const url = EndpointUtil.resolveProjectEndpoint(this.connection, this.id, endpoint as ProjectEndpoints);
+                let wStream;
+                try {
+                    wStream = fs.createWriteStream(profilingOutPath);
+                } catch (error) {
+                    Log.e(error);
+                    return;
+                }
+                try {
+                    await this.asyncHttpRequest(url, this.connection.url.scheme === "https", wStream);
+                } catch (error) {
+                    Log.e(error);
+                    vscode.window.showErrorMessage(`Error receiving profiling data for ${this.name}`);
+                } finally {
+                    wStream.close();
+                }
+                Log.d(`Saved profiling data to project ${this.name} at ${profilingOutPath}`);
+                break;
+            default:
+                break;
+        }
+    }
+
+    private asyncHttpRequest(url: string, secure: boolean = false, wStream: fs.WriteStream): Promise<void> {
```

Comment:
This utility function shouldn't be in the Project class, move it into `Requester`